datameta/api/metadatasets.py

```diff
@@ -136,6 +166,136 @@ def delete_metadatasets(request: Request) -> HTTPNoContent:
     return HTTPNoContent()
 
 
+@view_config(
+    route_name="rpc_update_metadatasets",
+    renderer="json",
+    request_method="POST",
+    openapi=True
+)
+def update_metadatasets(request: Request) -> SubmissionResponse:
+    auth_user = security.revalidate_user(request)
+    db = request.dbsession
+
+    # Obtain string converted version of the record
+    record = record_to_strings(request.openapi_validated.body["record"])
+
+    replaces = request.openapi_validated.body["replaces"]
+    replaces_label = request.openapi_validated.body["replacesLabel"]
+
+    # Query the configured metadata. We're only considering and allowing
+    # non-service metadata when creating a new metadataset.
+    metadata = get_all_metadata(db, include_service_metadata=False)
+
+    # prevalidate (raises 400 in case of validation failure):
+    validate_metadataset_record(metadata, record)
+
+    # Render records according to MetaDatum constraints.
+    record = render_record_values(metadata, record)
+
+    # construct new MetaDataSet:
+    mdata_set = MetaDataSet(
+        site_id = siteid.generate(request, MetaDataSet),
+        user_id = auth_user.id,
+        submission_id = None
+    )
+
+    db.add(mdata_set)
+
+    mset_repl_evt = MsetReplacementEvent(
+        user_id = auth_user.id,
+        datetime = datetime.utcnow(),
+        label = replaces_label,
+        new_metadataset_id = mdata_set.id
```

Comment:
Needs a `flush()` before accessing `mdata_set.id`

Reply:
makes sense, but what happens when one of the downstream checks fails?

Reply:
i'll see if i can re-arrange this

Reply:
A flush is not a commit. The transaction will still be rolled back without any problems.

Reply:
i see, cheers.